### PR TITLE
feat: Adapt code to new architecture

### PR DESCRIPTION
**Pull request by egcastro**
Related PR: https://github.com/snyk/topcoat-core/pull/469

### CODE REVIEW

File: library/components/TDropdown.vue [hunk 1]
```diff
@@ -43,73 +39,88 @@
...
 			},
-			setPopup(toggle) {
-				this.popup = toggle;
-				this.internal_active = toggle;
-
-				if (toggle) {
-					document.body.addEventListener("click", this.handleOutsideClick);
-				} else {
-					document.body.removeEventListener("click", this.handleOutsideClick);
+			openPopup() {
```
Comment (sareg0) [inline library/components/TDropdown.vue:102]:
```suggestion
			togglePopup() {
```

Non-blocking for merging: only a suggestion to better capture that the function handles closing the popup as well as opening

File: visualizations/frontend/filters/v2/TMultiSelector.vue [hunk 2]
```diff
@@ -173,6 +170,9 @@
             updateUrlParam() {
                 this.setFilterValue("selected_items", this.checked.join('|'), true);
             },
+            onDropdownOpen() {
+                this.fetchLayerData();
+            }
```
Comment (sareg0) [inline visualizations/frontend/filters/v2/TMultiSelector.vue:175]: 💅 💪 ✨

File: library/components/TDropdown.vue [hunk 3]
```diff
@@ -43,73 +39,88 @@
...
-				})
+				});
 			},
 		},
+		watch: {
+			isExpanded: function(newVal, oldVal) {
+				if (newVal) {
+					this.isPopupOpen = true;
+					this.$emit('open');
+				}
```
Comment (sareg0) [inline library/components/TDropdown.vue:121]: Do we need an else to emit 'close'?

Comment (egcastro) [reply library/components/TDropdown.vue:121]: Not really. This is quite complicated to understand by the way we are wiring components with the render function (Vue's render function) and scoped slots. The trick here is that we can re-use the same code path that this component's parent will follow if the `@open` callback is been called when a props like `isExpanded` is change. But this is due to the fact that we are rendering components with this prop `is-expanded` set to true already. Meaning that there was never a `click` and no way to trigger the `fetchLayerData` callback programatically. This is a way of simulating a `click` programatically based on a prop set/change 😄

Comment (egcastro) [reply library/components/TDropdown.vue:121]: But the `close` is only trigger but the clicking outside. There is no way to pass a prop for closing the the dropdown.
This is mostly a bigger topic that we should discuss at some point, components' API and behaviours in general.

Comment (sareg0) [reply library/components/TDropdown.vue:121]: ah, ok. I thought there was probably a good reason, but felt compelled to ask.

**Review by gau94rav — APPROVED** [review]
LGTM.

**Pull request merged**